qemu/target-i386/cpu.h

```diff
@@ -699,7 +699,7 @@ typedef enum {
 
 typedef struct SegmentCache {
     uint32_t selector;
-    target_ulong base;
+    uint64_t base;   /* handle 32 or 64 bit CPUs */
```

Comment:
1. When I built unicorn on a 32 bit machine today, target_ulong ended up being 64 bits. I will restate my question from earlier, what determines the value of TARGET_LONG_BITS during the build process.

2. When you say "no, target_long is 32bit for 32bit machine, and 64bit for 64bit machine" do you mean host or guest? If target_ulong is 32 bits on a 32 bit host, then how do I emulate x86_64 on a 32 bit host since 64 bit register values will not fit into target_ulong.

3. We can't copy any fields from a void* argument to uc_reg_write unless we can cast that pointer to a known data structure type. This is why in my original pull request I had the registers broken out into UC_X86_REG_GDTR_LIMIT, UC_X86_REG_GDTR_BASE, etc